frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -85,6 +86,11 @@ const RestorePVCModal = withHandlePromise<RestorePVCModalProps>(
       PersistentVolumeClaimKind
     >(PersistentVolumeClaimModel, resource?.spec?.source?.persistentVolumeClaimName, namespace);
 
+    const [scResource, scResourceLoaded, scResourceLoadError] = useK8sGet<StorageClassResourceKind>(
+      StorageClassModel,
+      pvcResource?.spec?.storageClassName,
```

Comment:
check what `useK8sGet` returns when `pvcResource` is not loaded and `pvcResource?.spec?.storageClassName` is undefined...
we need to add proper condition in the loading states...

Reply:
or use `k8sList` and filter for the StorageClass later, that will be better in this case...

Reply:
I checked and the status are as follows:

When pvcResource is not loaded or undefined or pvcResource?.spec?.storageClassName is undefined: 
scResource became undefined and scResourceLoaded is false and scResourceLoadError is undefined.

Reply:
can u check this once:
```
const [scResource, scResourceLoaded, scResourceLoadError] = useK8sGet<StorageClassResourceKind>(StorageClassModel, null);
```
and print the values of `scResource, scResourceLoaded, scResourceLoadError` and let me know...

Reply:
> can u check this once:
> 
> ```
> const [scResource, scResourceLoaded, scResourceLoadError] = useK8sGet<StorageClassResourceKind>(StorageClassModel, null);
> ```
> 
> and print the values of `scResource, scResourceLoaded, scResourceLoadError` and let me know...

scResource = null, scResourceLoaded = false, scResourceLoadError= null

Reply:
Hope its taking care of it. So do we need to consider checking the stats of pvcResource before using them for scResource? or shall i simply leave it, any suggestion / could you please give some more light ...

Reply:
it will not return null, it will return list of all the StorageClasses, please check properly...

Reply:
check how `k8sGet` works...

Reply:
https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts#L55-L56